src/main/scala/util/TraceEncoderController.scala

```diff
@@ -0,0 +1,99 @@
+// See LICENSE.Berkeley for license details.
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.util
+
+import chisel3._
+import chisel3.util._
+import org.chipsalliance.cde.config.Parameters
+import org.chipsalliance.diplomacy.lazymodule._
+import freechips.rocketchip.diplomacy.{AddressSet}
+import freechips.rocketchip.resources.{SimpleDevice}
+import freechips.rocketchip.tilelink.TLRegisterNode
+import freechips.rocketchip.regmapper.{RegField, RegFieldDesc}
+
+class TraceEncoderControlInterface() extends Bundle {
+  val enable = Bool()
+  val target = UInt(TraceSinkTarget.getWidth.W)
+  val hpmcounter_enable = UInt(32.W)
+  val hpmcounter_report_interval = UInt(32.W)
+}
+class TraceEncoderController(addr: BigInt, beatBytes: Int)(implicit p: Parameters) extends LazyModule {
```

Comment:
Can this thing just be combined with the Arbiter? Since this thing sets the control signals for the arbiter.